src/main/scala/cromiam/samiam/SamIamActor.scala

```diff
@@ -0,0 +1,56 @@
+package cromiam.samiam
+
+import akka.actor.{Actor, ActorLogging, Props}
+import cromiam.samiam.SamIamActor._
+
+import scala.concurrent.ExecutionContext
+import scala.util.Random
+import scala.concurrent.duration._
+
+final class SamIamActor(allowedUsers: List[String]) extends Actor with ActorLogging {
+
+  implicit val ec: ExecutionContext = context.dispatcher
+
+  override def receive: Receive = {
+    case r: SamIamActorRequest => handleRequest(r)
+    case other => log.error(s"Unexpected message to SamIamActor: $other")
+  }
+
+  private def handleRequest: Function[SamIamActorRequest, Unit] = {
+    case RequestAuthorization(ar) =>
+      val respondTo = sender()
+      // TODO: Replace with a real IAM lookup:
+      val delay = (Random.nextInt(500) + 100).milliseconds
+      val message = if (allowedUsers.contains(ar.user)) UserActionAuthorized(ar) else UserActionDenied(ar)
+      context.system.scheduler.scheduleOnce(delay) { respondTo ! message }
+      ()
+    case RegisterWorkflow(u, w) =>
+      val respondTo = sender()
+      // TODO: Replace with a real IAM registration:
+      val delay = (Random.nextInt(500) + 100).milliseconds
+      context.system.scheduler.scheduleOnce(delay) { respondTo ! WorkflowRegistrationComplete(u, w) }
+      ()
+  }
+}
+
+object SamIamActor {
+  def props(allowedUsers: List[String]): Props = Props(new SamIamActor(allowedUsers))
+
+  sealed trait SamIamActorRequest
+  final case class RequestAuthorization(authorizationRequest: AuthorizationRequest) extends SamIamActorRequest
+  final case class RegisterWorkflow(user: String, workflowId: String) extends SamIamActorRequest
+
+  sealed trait SamIamActorResponse
+  final case class UserActionAuthorized(authorizationRequest: AuthorizationRequest) extends SamIamActorResponse
+  final case class UserActionDenied(authorizationRequest: AuthorizationRequest) extends SamIamActorResponse
+  final case class WorkflowRegistrationComplete(user: String, workflowId: String) extends SamIamActorResponse
+
+  sealed trait AuthorizationRequest {
+    def user: String
+    def action: String
+  }
+  final case class WorkflowAuthorizationRequest(user: String, workflowId: String, action: String) extends AuthorizationRequest
+  final case class CreateWorkflowAuthorizationRequest(user: String) extends AuthorizationRequest {
+    override val action = "create"
```

Comment:
If I understand what's going on here I think I've heard this come up as a request, I remember having a similar "news to me" reaction as you here. I would have said that it was you who mentioned it but since you don't seem to be that person perhaps it was @kcibul.